File: src/python/p_0027_remove_element/p_0027_remove_element.py
from typing import List


class Solution:
    def removeElement(self, nums: List[int], val: int) -> int:
        i, j = 0, len(nums) - 1
        while i <= j:
            if nums[i] == val:
                nums[i] = nums[j]
                del nums[j]
                j -= 1
            else:
                i += 1

        return j + 1


solution = Solution()

solution.removeElement([1], 1)

File: src/typescript/p_0027_remove_element/p_0027_remove_element.ts
/* time: O(n), space: O(1) */
export function removeElement(nums: number[], val: number): number {
    let leftIndex = 0;
    let rightIndex = nums.length - 1;

    while (leftIndex <= rightIndex) {
        if (nums[leftIndex] === val) {
            nums[leftIndex] = nums[rightIndex];
            rightIndex--;
        } else {
            leftIndex++;
        }
    }

    return leftIndex;
}
